NumSet/NumSet/Writer/SetWriter.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Assingment4.Writer
+{
+    public class SetWriter : IDisposable
+    {
+        private bool _DisposedValue;
+
+        private string? _filePath;
+        public string? FilePath { get => _filePath; set => _filePath = value; }
+
+        private StreamWriter? fileWriter;
+        public StreamWriter? FileWriter { get => fileWriter; set => fileWriter = value; }
```

Comment:
You don't need a private variable and a public auto property. You only need one or the other. In this case, you really only want the private variable. Just make sure to name it properly using PascalCase. Also, you don't need to make it nullable since you will be creating it in the constructor.
```suggestion
        private StreamWriter _FileWriter;
```